contracts/SwapAndBridge.sol

```diff
@@ -291,4 +328,178 @@ contract UniversalSwapAndBridge is SwapAndBridgeBase {
             acrossInputToken
         );
     }
+
+    /**
+     * @notice Swaps an EIP-2612 token on this chain via specified router before submitting Across deposit atomically.
+     * Caller can specify their slippage tolerance for the swap and Across deposit params.
+     * @dev If swapToken does not implement `permit` to the specifications of EIP-2612, this function will fail.
```

Comment:
Is this comment still correct in light of the try/catch wrapper?

Reply:
It is. This comment is a bit more general, and just saying that you can't expect permit to work for a token which doesn't implement permit.